File: frontend/src/api/providers.ts
import { apiClient } from './client'

export interface Provider {
  provider_id: string
  name: string
  version: string
  description: string
  supported_services: string[]
  capabilities: string[]
  authentication_methods: string[]
}

export interface ProvidersResponse {
  success: boolean
  data: Provider[]
  metadata: {
    total: number
    generated_at: string
    api_version: string
  }
}

export interface Capability {
  provider_id: string
  capabilities: string[]
  supported_services: string[]
  supported_collectors: string[]
  supported_normalizers: string[]
}

export interface CapabilitiesResponse {
  success: boolean
  data: Capability[]
  metadata: {
    total: number
    generated_at: string
    api_version: string
  }
}

export async function getProviders(): Promise<ProvidersResponse> {
  const response = await apiClient.get('/providers')
  return response.data
}

export async function getCapabilities(): Promise<CapabilitiesResponse> {
  const response = await apiClient.get('/capabilities')
  return response.data
}

File: backend/api/routers/providers.py
from fastapi import APIRouter, Depends
from backend.api.dependencies import get_plugin_manager
from backend.plugins.manager import PluginManager


router = APIRouter()


@router.get(
    "/providers",
    response_model=dict,
    tags=["Providers"],
    summary="List registered providers",
    description="Returns all registered cloud provider plugins with their metadata.",
)
def list_providers(plugin_manager: PluginManager = Depends(get_plugin_manager)):
    providers = []
    for provider_id in plugin_manager.list_plugins():
        metadata = plugin_manager.get_metadata(provider_id)
        providers.append({
            "provider_id": metadata.provider_id,
            "name": metadata.name,
            "version": metadata.version,
            "description": metadata.description,
            "supported_services": metadata.supported_services,
            "capabilities": metadata.capabilities,
            "authentication_methods": metadata.authentication_methods,
        })
    return {
        "success": True,
        "data": providers,
        "metadata": {
            "total": len(providers),
            "generated_at": __import__('datetime').datetime.utcnow().isoformat(),
            "api_version": "v1",
        },
    }
